### PR TITLE
Automatically initialize title tooltips after loading a report

### LINKED ISSUE #12676: Tooltips in conversion overview not working properly
Maybe some other reports are affected as well. Maybe we need a global tooltip initialization that is triggered after reports are loaded

image

### PR DESCRIPTION
**Pull request by sgiehl**
not sure if there would be a better place to handle that, but at least it seems to work this way ;-)

fixes #12676

### CODE REVIEW

File: plugins/CoreHome/angularjs/widget/widget.directive.js [hunk 1]
```diff
@@ -96,6 +96,16 @@
...
+
+                    $(element).tooltip({
+                        track: true,
+                        content: function() {
+                            var title = $(this).attr('title');
+                            return piwikHelper.escape(title.replace(/\n/g, '<br />'));
+                        },
+                        show: false,
+                        hide: false
+                    });
```
Comment (diosmosis) [inline plugins/CoreHome/angularjs/widget/widget.directive.js]: Should we also destroy the tooltip when the scope is destroyed?

**Review by diosmosis — COMMENTED** [review]
Appears to work, tested tooltips in datatable, visitorlog, etc., existing tooltips work, and other places have tooltips now. Left one minor comment.

Actually, one possible idea, though not sure how good it is. An alternative solution might be to make the `title=` attribute an angular directive that uses `$(element).tooltip()`. This might be better than calling it on a root element, which I believe would have to select each sub element w/ a title? Nah, that's probably not really better. I think the solution in the PR is good 👍

**Review by diosmosis — APPROVED** [review]

**Comment by diosmosis** [comment]
There's a UI test failing: https://builds-artifacts.matomo.org/matomo-org/matomo/3.x-dev/27273/UIIntegrationTest_segmented_visitorlog.png

looks like the row in visitor logs shouldn't have a tooltip?

**Comment by sgiehl** [comment]
Tests work again...

**Pull request merged**